src/mem/impls.rs

```diff
@@ -41,6 +41,65 @@ unsafe fn read_usize_unaligned(x: *const usize) -> usize {
     core::mem::transmute(x_read)
 }
 
+/// Load `load_sz` many bytes from `src`, which must be usize-aligned. Acts as if we did a `usize`
+/// read with the out-of-bounds part filled with 0s.
+/// `load_sz` be strictly less than `WORD_SIZE`.
+#[cfg(not(feature = "mem-unaligned"))]
+#[inline(always)]
+unsafe fn load_aligned_partial(src: *const usize, load_sz: usize) -> usize {
+    debug_assert!(load_sz < WORD_SIZE);
+
+    let mut i = 0;
+    let mut out = 0usize;
+    macro_rules! load_prefix {
+        ($($ty:ty)+) => {$(
+            let chunk_sz = core::mem::size_of::<$ty>();
+            if (load_sz & chunk_sz) != 0 {
+                // Since we are doing the large reads first, this must still be aligned to `chunk_sz`.
+                *(&raw mut out).wrapping_byte_add(i).cast::<$ty>() = *src.wrapping_byte_add(i).cast::<$ty>();
+                i |= chunk_sz;
+            }
+        )+};
+    }
```

Comment:
Could this be a function shared between `load_aligned_partial` and `load_aligned_end_partial` rather than a macro? A la

```rust
pub unsafe fn load_aligned_impl<T: Copy>(
    src: *const usize,
    out: &mut usize,
    offset: usize,
    load_sz: usize,
) -> usize {
    let chunk_sz = core::mem::size_of::<T>();
    if (load_sz & chunk_sz) != 0 {
        unsafe {
            *core::ptr::from_mut(out)
                .wrapping_byte_add(offset)
                .cast::<T>() = *src.wrapping_byte_add(offset).cast::<T>();
        }
        offset | chunk_sz
    } else {
        offset
    }
}
```